lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/UpdateTaskHost.cs

```diff
@@ -148,27 +155,80 @@ public byte[] Call(byte[] memento)
                     "Received OperationCanceledException in UpdateTaskHost with message: {0}.",
                     e.Message);
             }
-            catch (IOException e)
+            catch (Exception e)
             {
-                Logger.Log(Level.Error, "Received IOException in UpdateTaskHost with message: {0}.", e.Message);
-                if (!_cancellationSource.IsCancellationRequested)
+                if (e is IOException || e is TcpClientConnectionException || e is RemotingException ||
```

Comment:
"A common source of cascading failures in distributed systems is an attempt to require a lot of resources at once. In this PR, what would happen if an IMRU job with a large number of mappers suddenly loses half of them and then tries to resubmit them all at once ? I believe this case deserves a special test with YARN."  -- we have a configurable percentage for number of failures allowed.